implants/golem/tests/cli.rs

```diff
@@ -0,0 +1,61 @@
+use assert_cmd::prelude::*; // Add methods on commands
+use predicates::prelude::*; // Used for writing assertions
+use std::process::Command; // Run programs
+
+// Test running `./golem ./working_dir/tomes/hello_world.tome`
+#[test]
+fn test_golem_main_basic_non_interactive() -> anyhow::Result<()> {
+    let mut cmd = Command::cargo_bin("golem")?;
+
+    cmd.arg("working_dir/tomes/hello_world.tome");
+    cmd.assert()
+        .success()
+        .stdout(predicate::str::contains("[\"HELLO\"]"));
+
+    Ok(())
+}
+
+
+// Test running `./golem ./working_dir/tomes/eldritch_test.tome`
+#[test]
+fn test_golem_main_basic_eldritch_non_interactive() -> anyhow::Result<()> {
+    let mut cmd = Command::cargo_bin("golem")?;
+
+    cmd.arg("working_dir/tomes/eldritch_test.tome");
+    cmd.assert()
+        .success()
+        .stdout(predicate::str::contains("[\"[\\\"append\\\", \\\"copy\\\", \\\"download\\\", \\\"exists\\\", \\\"hash\\\", \\\"is_dir\\\", \\\"is_file\\\", \\\"mkdir\\\", \\\"read\\\", \\\"remove\\\", \\\"rename\\\", \\\"replace\\\", \\\"replace_all\\\", \\\"timestomp\\\", \\\"write\\\"]\"]"));
+
+    Ok(())
+}
+
+
+// Test running `echo "dir(file)" | ./golem` for interactive mode.
+#[test]
+fn test_golem_main_basic_interactive() -> anyhow::Result<()> {
```

Comment:
Still trying to figure out the right way to test an interactive program in rust.
I was able to run system commands using `Command::new()` however since golem is being built the system can't find it.
It seems that I'm stuck using the `Command::cargo_bin()` function however that doesn't seem to expose the stdin variable for me to pass data in.
Any suggestions would be appreciated 🙏